### ISSUE TITLE
[Business Approvals Kit - BUG] Custom Connector Test Connection Issue

### ISSUE THREAD

**Issue by fkopan82** [microsoft/powercat-business-approvals-kit#162]
### Describe the issue

While editing the Approvals kit custom connector, I am unable to complete a Test operation for the "GetApprovalDataFields" operation (under the Test tab). After I click on field "New connection" and pick an available user account from the provided list, the field is not updated with the selected connection, hence "None" value remains frozen.
![Connector Test](https://github.com/microsoft/powercat-business-approvals-kit/assets/63752634/416e6a52-520e-486a-ac4c-7f9f6a92a1d7)

### Expected Behavior

Expected to complete the Test connection for the selected operation "GetApprovalDataFields".

### What solution version are you using?

1.0.20231214.1

### What app or flow are you having the issue with?

Business Approval Kit | O365 Users

AB#1837

**Comment by Ramakrishnan24689** [comment]
@fkopan82, Can you confirm what happens if you create new connection? Since this is a custom connector connection that needs to be created, you will have to create a new connection for the first time if havn't already.

Can you also confirm, if you have followed the suggested step mentioned in [here](https://github.com/microsoft/powercat-business-approvals-kit/issues/155#issuecomment-1862099905)

**Comment by fkopan82** [comment]
Hi,
When I click New connection, the attached Sign In screen shows, on which I select my MS user account. After select, nothing happens on screen.
![New connection](https://github.com/microsoft/powercat-business-approvals-kit/assets/63752634/50947775-23ae-494c-bc0b-006e3ab83dab)
In the Setup guide section (https://learn.microsoft.com/en-us/power-automate/guidance/business-approvals-templates/setup) I did not found instructions regarding update of Host URL in the General Tab.
Can you please instruct what needs to be done here?

Best regards,
Filip

**Comment by Grant-Archibald-MS** [comment]
@fkopan82 the setup page needs an update to include the additional settings of the organization url:

The setup notes from the release notes include some guidance in https://github.com/microsoft/powercat-business-approvals-kit/releases/tag/BusinessApprovalKit-December2023

> [!IMPORTANT]
> In the Update the Approvals kit with a custom connector section of the setup guide the additional notes apply:
> 1. In the General Tab enter the host name the environment that the Approvals Kit has been installed. This should be in a format similar to yourorg.crm.dynamics.com. Read https://github.com/microsoft/powercat-business-approvals-kit/issues/135 https://github.com/microsoft/powercat-business-approvals-kit/issues/135
> 2. You can obtain the organization url from https://make.powerapps.com/ settings [Read how](https://learn.microsoft.com/power-apps/maker/canvas-apps/get-sessionid#get-the-session-id-for-power-apps-makepowerappscom)

**Comment by fkopan82** [comment]
Hi,

Amended the custom connector as described but the issue is still present.
Can you please confirm what needs to be entered Security Tab / Resource URL? Is this the same as Host URL?

Many thanks.

Best regards,
Filip

**Comment by Ramakrishnan24689** [comment]
> Hi, Amended the custom connector as described but the issue is still present. Can you please confirm what needs to be entered Security Tab / Resource URL? Is this the same as Host URL? Many thanks. Best regards, Filip 

@fkopan82, Thanks for reaching out.

Let say your environment url is **https://orgc18dc7f1.crm.dynamics.com/** for example,
then Resource URL will be **https://orgc18dc7f1.crm.dynamics.com**
and Host Url will be **orgc18dc7f1.crm.dynamics.com**

In case you don't know the Environment URL, follow the below steps which are one of the ways to find it

1) Login to make.powerapps.com and select your environment from option on top left global navigation.

![image](https://github.com/microsoft/powercat-business-approvals-kit/assets/11135056/adfeef2d-3f7b-45ea-9d40-1fdee3195cf0)
2) Click on the gear icon and select Advanced settings.
![image](https://github.com/microsoft/powercat-business-approvals-kit/assets/11135056/7dce814c-186c-4722-a4e4-00372e023d40)


3) You will be redirected to a new tab. Copy the URL of the tab which is the environment URL.

![image](https://github.com/microsoft/powercat-business-approvals-kit/assets/11135056/83482987-f2d1-4036-be5a-22f98dc968d2)

Let us know, if the issue persists even after updating the connector with the Resource and Host URL.

**Comment by fkopan82** [comment]
Hi,

Thanks for the support.
Amending the Resource URL as per the format below solved the issue (maybe this should be noted in the setup instructions in order to avoid any confusion).

However after testing the connection , error 400 is returned with message : "The parameter 'ProcessID=undefined' is not in scope." .

Should this be raised as a separate issue?

Best regards

**Comment by Ramakrishnan24689** [comment]
> Hi, Thanks for the support. Amending the Resource URL as per the format below solved the issue (maybe this should be noted in the setup instructions in order to avoid any confusion). However after testing the connection , error 400 is returned with message : "The parameter 'ProcessID=undefined' is not in scope." 

Yes, we will work updating the documentation to highlight this more clearly.
From your current error message - "_The parameter 'ProcessID=undefined' is not in scope._", it appears you are testing **GetApprovalDataFields** operation which is not part of the test mentioned in the [documentation](https://review.learn.microsoft.com/en-us/power-automate/guidance/business-approvals-templates/setup?branch=main#update-the-approvals-kit-with-a-custom-connector)

It is suggested to test **GetPublishedWorkflow** operation instead.

![image](https://github.com/microsoft/powercat-business-approvals-kit/assets/11135056/f80b55fd-3e21-485c-992a-38712a0d7422)

**Comment by fkopan82** [comment]
Yes correct, I overlooked this detail, it ok now.
Many thanks.

Filip

**Comment by Grant-Archibald-MS** [comment]
@fkopan82 I have added https://github.com/microsoft/powercat-business-approvals-kit/blob/main/Workshop/docs/connector-setup.md to give some additional context.

We will look at updating the setup guide on learn to merge in this content

**Comment by maggiepei123** [comment]
Hi all,
I am tring to update the Approvals Kit connector to BusinessApprovalKit_1.0.20240402.1_managed.
After I completed all of the previous setting, I found the message showed 
Operation failed (401),
The respond header is {
  "allow": "OPTIONS,GET,HEAD,POST",
  "authactivityid": "f9f18f03-a043-404a-adcb-0af6a85ba46c",
  "content-length": "0",
  "crm.serviceid": "CRMAppPool",
  "date": "Wed, 24 Apr 2024 02:27:27 GMT",
  "nativewebsession-version": "2",
  "public": "OPTIONS,GET,HEAD,POST",
  "req_id": "0923343b-62ad-48dd-8715-11d6acc09093",
  "strict-transport-security": "max-age=31536000; includeSubDomains",
  "www-authenticate": "Bearer authorization_uri=https://login.microsoftonline.com/67c96cd6-4a95-46ee-ac9b-8ee13ae256c4/oauth2/authorize, resource_id=https://orgc08dc7f0.crm.dynamics.com/",
  "x-ms-apihub-cached-response": "true",
  "x-ms-apihub-obo": "false",
  "x-ms-service-request-id": "e61e976b-3f71-471b-bfbf-392adaf7c130,0923343b-62ad-48dd-8715-11d6acc09093",
  "x-source": "271331701671364312419241981242252051127711220928212136235121631412351302042201692231636,1561668611229674753511017921251854217713920512587222203137521177312311515236587"
}
What should I do? I am looking forward to your help.
![Screenshot 2024-04-24 103130](https://github.com/microsoft/powercat-business-approvals-kit/assets/165247682/f0fb6ad1-0273-475c-9cb1-46fb1f548da5)
![Screenshot 2024-04-24 103200](https://github.com/microsoft/powercat-business-approvals-kit/assets/165247682/87dcb0a5-f05b-47a6-b38a-9c30baf466c8)
